src/helpers/amoc_xmpp_handlers.erl

```diff
@@ -57,12 +57,27 @@ stanza_handler(Pred, Action) ->
             end
     end.
 
+check_predicate(Pred, Stanza) when is_function(Pred, 1)->
+    apply_safely(Pred, [Stanza]).
```

Comment:
What is the motivation for all the 'apply_safely'? If the predicate crashes or the handler crashes, the whole client process should crash as well because there is a bug in the scenario. I am afraid that we might miss some bugs otherwise. For example, I found one or two bugs in my handlers because I saw that some clients disconnected unexpectedly. I really need this to crash.